### ISSUE TITLE
perl-5.8.3 File::Find (find.t) fails tests

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#7197]
Migrated from [rt.perl.org#27923](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=27923) (status was 'resolved')

Searchable as RT27923$

**Comment by p5pRT** [comment]
### From lanny@cisco.com

#### Created by lanny@cisco.com



Built perl\-5\.8\.3 on RedHat Enterprise version 3\.  File&#8203;::Find \(find\.t\)
failed on C\<make test>\.

bash\-2\.05b$ \./perl harness \.\./lib/File/Find/t/find\.t
\.\./lib/File/Find/t/find\.\.\.\.FAILED tests 161\, 173\, 187\-189                    
&nbsp;        Failed 5/189 tests\, 97\.35% okay
Failed Test               Stat Wstat Total Fail  Failed  List of Failed
\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
\.\./lib/File/Find/t/find\.t              189    5   2\.65%  161 173 187\-189
Failed 1/1 test scripts\, 0\.00% okay\. 5/189 subtests failed\, 97\.35% okay\.

The tests seem to fail as File&#8203;::Find follows a soft\-link\.

To double check myself I built perl\-5\.8\.3 on Solaris 2\.8 using same
options \(as much as possible\) and C\<make test> passed without errors\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl v5.8.3:

Configured by lanny at Wed Mar 24 13:22:43 PST 2004.

Summary of my perl5 (revision 5.0 version 8 subversion 3) configuration:
  Platform:
    osname=linux, osvers=2.4.21-9.elsmp, archname=i686-linux
    uname='linux linux-build9.cisco.com 2.4.21-9.elsmp #1 smp thu jan 8 17:08:56 est 2004 i686 i686 i386 gnulinux '
    config_args='-Dmksymlinks -Dprefix=/usr/cisco -Dinstallprefix=/usr/cisco/packages/perl/perl-5.8.3'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-fno-strict-aliasing -DDEBUGGING -I/usr/cisco/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm',
    optimize='-O',
    cppflags='-fno-strict-aliasing -DDEBUGGING -I/usr/cisco/include -I/usr/include/gdbm'
    ccversion='', gccversion='3.2.3 20030502 (Red Hat Linux 3.2.3-24)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =''
    libpth=/usr/cisco/lib /lib /usr/lib
    libs=-lnsl -lgdbm -ldb -ldl -lm -lcrypt -lutil -lc
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lc
    libc=/lib/libc-2.3.2.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.3.2'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared'

Locally applied patches:
    


@INC for perl v5.8.3:
    ../lib
    /usr/cisco/lib/perl5/5.8.3/i686-linux
    /usr/cisco/lib/perl5/5.8.3
    /usr/cisco/lib/perl5/site_perl/5.8.3/i686-linux
    /usr/cisco/lib/perl5/site_perl/5.8.3
    /usr/cisco/lib/perl5/site_perl
    .


Environment for perl v5.8.3:
    HOME (unset)
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/bin:/bin
    PERL_BADLANG (unset)
    SHELL (unset)

```
</details>

**Comment by p5pRT** [comment]
### From accounts@melbourneit.com.au

#### Created by lanny@cisco.com



Built perl\-5\.8\.3 on RedHat Enterprise version 3\.  File&#8203;::Find \(find\.t\)
failed on C\<make test>\.

bash\-2\.05b$ \./perl harness \.\./lib/File/Find/t/find\.t
\.\./lib/File/Find/t/find\.\.\.\.FAILED tests 161\, 173\, 187\-189                    
&nbsp;        Failed 5/189 tests\, 97\.35% okay
Failed Test               Stat Wstat Total Fail  Failed  List of Failed
\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
\.\./lib/File/Find/t/find\.t              189    5   2\.65%  161 173 187\-189
Failed 1/1 test scripts\, 0\.00% okay\. 5/189 subtests failed\, 97\.35% okay\.

The tests seem to fail as File&#8203;::Find follows a soft\-link\.

To double check myself I built perl\-5\.8\.3 on Solaris 2\.8 using same
options \(as much as possible\) and C\<make test> passed without errors\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl v5.8.3:

Configured by lanny at Wed Mar 24 13:22:43 PST 2004.

Summary of my perl5 (revision 5.0 version 8 subversion 3) configuration:
  Platform:
    osname=linux, osvers=2.4.21-9.elsmp, archname=i686-linux
    uname='linux linux-build9.cisco.com 2.4.21-9.elsmp #1 smp thu jan 8 17:08:56 est 2004 i686 i686 i386 gnulinux '
    config_args='-Dmksymlinks -Dprefix=/usr/cisco -Dinstallprefix=/usr/cisco/packages/perl/perl-5.8.3'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-fno-strict-aliasing -DDEBUGGING -I/usr/cisco/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm',
    optimize='-O',
    cppflags='-fno-strict-aliasing -DDEBUGGING -I/usr/cisco/include -I/usr/include/gdbm'
    ccversion='', gccversion='3.2.3 20030502 (Red Hat Linux 3.2.3-24)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =''
    libpth=/usr/cisco/lib /lib /usr/lib
    libs=-lnsl -lgdbm -ldb -ldl -lm -lcrypt -lutil -lc
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lc
    libc=/lib/libc-2.3.2.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.3.2'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared'

Locally applied patches:
    


@INC for perl v5.8.3:
    ../lib
    /usr/cisco/lib/perl5/5.8.3/i686-linux
    /usr/cisco/lib/perl5/5.8.3
    /usr/cisco/lib/perl5/site_perl/5.8.3/i686-linux
    /usr/cisco/lib/perl5/site_perl/5.8.3
    /usr/cisco/lib/perl5/site_perl
    .


Environment for perl v5.8.3:
    HOME (unset)
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/bin:/bin
    PERL_BADLANG (unset)
    SHELL (unset)




```
</details>

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @iabyn

On Wed\, Mar 24\, 2004 at 09&#8203;:52&#8203;:19PM \-0000\, lanny@&#8203;cisco\.com \(via RT\) wrote&#8203;:
> \# New Ticket Created by  lanny@&#8203;cisco\.com 
> failed on C\<make test>\.
> 
> bash\-2\.05b$ \./perl harness \.\./lib/File/Find/t/find\.t
> \.\./lib/File/Find/t/find\.\.\.\.FAILED tests 161\, 173\, 187\-189                    
>         Failed 5/189 tests\, 97\.35% okay
> Failed Test               Stat Wstat Total Fail  Failed  List of Failed
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \.\./lib/File/Find/t/find\.t              189    5   2\.65%  161 173 187\-189
> Failed 1/1 test scripts\, 0\.00% okay\. 5/189 subtests failed\, 97\.35% okay\.
> 
> The tests seem to fail as File&#8203;::Find follows a soft\-link\.

Is there anything unusally about the build directory\, or the filesystem
on which it resides? eg is it mounted from the network\, or of an unusual
filesystem type?

\-\- 
"I do not resent critisism\, even when\, for the sake of emphasis\,
it parts for the time with reality"\.
&nbsp;    \-\- Winston Churchill\, House of Commons\, 22nd Jan 1941\.

**Comment by p5pRT** [comment]
### From lanny@cisco.com

I'm building in /tmp which is using tmpfs\.  I'll rebuild in a 
different fs type\.

Nice call\.  All tests successful\.

I'm happy with the result\.  Feel free to mark the case as closed\.

&nbsp;   \-ljr

PS \- I had the perl src still in tmp but the build directory via 
\-Dmksymlinks was on a ext3 filesystem\.

Dave Mitchell via RT wrote&#8203;:
> On Wed\, Mar 24\, 2004 at 09&#8203;:52&#8203;:19PM \-0000\, lanny@&#8203;cisco\.com \(via RT\) wrote&#8203;:
> 
>>\# New Ticket Created by  lanny@&#8203;cisco\.com 
>>failed on C\<make test>\.
>>
>>bash\-2\.05b$ \./perl harness \.\./lib/File/Find/t/find\.t
>>\.\./lib/File/Find/t/find\.\.\.\.FAILED tests 161\, 173\, 187\-189                    
>>        Failed 5/189 tests\, 97\.35% okay
>>Failed Test               Stat Wstat Total Fail  Failed  List of Failed
>>\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
>>\.\./lib/File/Find/t/find\.t              189    5   2\.65%  161 173 187\-189
>>Failed 1/1 test scripts\, 0\.00% okay\. 5/189 subtests failed\, 97\.35% okay\.
>>
>>The tests seem to fail as File&#8203;::Find follows a soft\-link\.
> 
> 
> Is there anything unusally about the build directory\, or the filesystem
> on which it resides? eg is it mounted from the network\, or of an unusual
> filesystem type?
> 

\-\- 
Lanny Ripple \<lanny@&#8203;cisco\.com>
IT\-ECS / Cisco Systems\, Inc\.

**Comment by p5pRT** [comment]
### From @nwc10

On Tue\, Mar 30\, 2004 at 10&#8203;:33&#8203;:53AM \-0600\, Lanny Ripple wrote&#8203;:
> I'm building in /tmp which is using tmpfs\.  I'll rebuild in a 
> different fs type\.
> 
> Nice call\.  All tests successful\.
> 
> I'm happy with the result\.  Feel free to mark the case as closed\.

There seems to be a real bug here\. File&#8203;::Find's tests are making unwarranted
assumptions about the order of results returned from readdir\.
We just got access to an x86\_64 machine this afternoon and it's failing
File&#8203;::Find's tests; I think entirely down to readdir returning directory
entries in another order\.

I intend to fix the test\.


Nicholas Clark

**Comment by p5pRT** [comment]
### From @iabyn

Fixed in bleedperl by Nicholas\, change \#22615; integrated into
5\.8\.4\-tobe by change 22623\.

**Comment by p5pRT** [comment]
@iabyn - Status changed from 'open' to 'resolved'
